nrpy/infrastructures/BHaH/BHaH_defines_h.py

```diff
@@ -97,7 +114,9 @@ def register_BHaH_defines(module: str, BHaH_defines: str) -> None:
 def output_BHaH_defines_h(
     project_dir: str,
     additional_includes: Optional[List[str]] = None,
-    enable_simd: bool = True,
+    restrict_pointer_type: str = "*restrict",
```

Comment:
To be revisited: setting a "RESTRICT" macro that does the appropriate thing.

Reply:
> To be revisited: setting a "RESTRICT" macro that does the appropriate thing.

Food for future thought:
While a macro can be used to resolve attribute inconsistencies between C and C++ (e.g. `restrict` vs `__restrict__`) which is the current solution that is used (`#define restrict __restrict__` in the cuda infrastructure), it wouldn't help at this level since the struct definitions cannot have a `restrict` argument at all and, thus, `__restrict__` can only be used as a function or pointer attribute where a malloc is not needed.

e.g.
`REAL *restrict params = grid[idx]->params;` 

is still valid as is

`f(REAL *restrict params)`